plugin/src/test/java/org/opensearch/ml/rest/RestMLDeleteModelGroupActionIT.java

```diff
@@ -6,32 +6,54 @@
 package org.opensearch.ml.rest;
 
 import java.io.IOException;
-import java.util.Map;
 
-import org.apache.hc.core5.http.HttpEntity;
-import org.junit.Ignore;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.rules.ExpectedException;
 import org.opensearch.client.Response;
 import org.opensearch.core.rest.RestStatus;
+import org.opensearch.ml.common.transport.model_group.MLRegisterModelGroupInput;
 import org.opensearch.ml.utils.TestHelper;
 
 public class RestMLDeleteModelGroupActionIT extends MLCommonsRestTestCase {
     @Rule
     public ExpectedException exceptionRule = ExpectedException.none();
+    private MLRegisterModelGroupInput mlRegisterModelGroupInput;
+    private String modelGroupId;
+
+    @Before
+    public void setup() throws IOException {
+        mlRegisterModelGroupInput = MLRegisterModelGroupInput.builder().name("testGroupID").description("This is test Group").build();
+        registerModelGroup(client(), TestHelper.toJsonString(mlRegisterModelGroupInput), registerModelGroupResult -> {
+            this.modelGroupId = (String) registerModelGroupResult.get("model_group_id");
+        });
+    }
 
-    @Ignore
     public void testDeleteModelGroupAPI_Success() throws IOException {
-        Response trainModelGroupResponse = ingestModelData();
-        HttpEntity entity = trainModelGroupResponse.getEntity();
-        assertNotNull(trainModelGroupResponse);
-        String entityString = TestHelper.httpEntityToString(entity);
-        Map map = gson.fromJson(entityString, Map.class);
-        String model_group_id = (String) map.get("model_group_id");
-
-        Response deleteModelResponse = TestHelper
-            .makeRequest(client(), "DELETE", "/_plugins/_ml/model_groups/" + model_group_id, null, "", null);
-        assertNotNull(deleteModelResponse);
-        assertEquals(RestStatus.OK, TestHelper.restStatus(deleteModelResponse));
+
+        Response deleteModelGroupResponse = TestHelper
+            .makeRequest(client(), "DELETE", "/_plugins/_ml/model_groups/" + modelGroupId, null, "", null);
+        assertNotNull(deleteModelGroupResponse);
+        assertEquals(RestStatus.OK, TestHelper.restStatus(deleteModelGroupResponse));
+    }
+
+    public void testDeleteAndRegisterModelGroup_Success() throws IOException, InterruptedException {
+
+        Response deleteModelGroupResponse = TestHelper
+            .makeRequest(client(), "DELETE", "/_plugins/_ml/model_groups/" + modelGroupId, null, "", null);
+
+        Thread.sleep(50);
```

Comment:
Will remove the sleep but when tested in local, with or without sleep, the same test was failing when IMMEDIATE refresh policy is not set. I gave minimum sleep to allow a brief pause to ensure the delete operation is processed right before register model group to avoid any flaky failures.